### PR TITLE
Improve location of ERR_ParamUnassigned error for local functions.

### LINKED ISSUE #12467: Squiggles are on the wrong span for out parameter unassigned error
We select the wrong span for this error: "out parameter must be assigned to before leaves the current method"...

This causes:
image

which makes it really hard to finish writing my method when I have to look through all the squiggles.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #12467.

@dotnet/roslyn-compiler, @agocke Please review.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs [hunk 1]
```diff
@@ -1606,5 +1606,139 @@ static void Main()
...
+    {
+        void TakeOutParam1(out int x)
+        {
+        }
+
+        int y;
+        TakeOutParam1(out y);
+    }
+
+        void TakeOutParam2(out int x)
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs]: Indentation

**Comment by jcouv** [comment]
LGTM

File: src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs (hunk 1 above)
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs]: This is a string literal. The indentation is intentional to keep two declarations aligned, makes it easier to verify consistency of error locations..

**Comment by AlekseyTs** [comment]
Ping. @dotnet/roslyn-compiler, @agocke Please review, need a second sign-off.

**Comment by cston** [comment]
LGTM

**Comment by agocke** [comment]
LGTM, thanks for the fix @AlekseyTs!

**Pull request merged**